mmdt/polls/views.py

```diff
@@ -68,4 +77,33 @@ def vote(request):
         except (KeyError, Choice.DoesNotExist, ValueError) as e:
             # Display an error message and redirect back to the index page
             messages.error(request, str(e))
-            return HttpResponseRedirect(reverse('polls:index') + f"?page={page}")
\ No newline at end of file
+            return HttpResponseRedirect(reverse('polls:index') + f"?page={page}")
+
+    @user_passes_test(lambda u: u.is_staff, login_url=None)  # Allow only staff members (admins)
+    def all_results(request):
+        # Fetch all questions with is_enabled=True for admins
+        all_questions = Question.objects.filter(is_enabled=True).order_by('poll_group', '-pub_date')
+
+        # If the user is not an admin, filter based on additional conditions
+        if not request.user.is_staff:
+            # Check if results are released
+            active_group = ActiveGroup.objects.filter(is_results_released=True).first()
+            if not active_group:
+                return HttpResponseForbidden("Results are not released yet.")
+
+            # Fetch questions related to the active group with additional conditions
+            all_questions = all_questions.filter(
+                poll_group=active_group,
+                poll_group__is_active=True
```

Comment:
I am not sure that filter work wells or not?

Reply:
is_results_released is only one group 
<img width="1667" alt="Screenshot 2024-02-26 at 4 19 19 PM" src="https://github.com/khinezarthwe/mmdt_web_app/assets/5201124/e96ba994-1040-4894-a3b4-659eb54d6545">


Question related to that group have only one question. 
<img width="1181" alt="Screenshot 2024-02-26 at 4 19 48 PM" src="https://github.com/khinezarthwe/mmdt_web_app/assets/5201124/e0ca6612-70da-4a68-93ab-b9200de388f3">


Actual output should be one question request release in view result page, but it shows all results.
<img width="1182" alt="Screenshot 2024-02-26 at 4 20 26 PM" src="https://github.com/khinezarthwe/mmdt_web_app/assets/5201124/b6c333f9-d865-40d8-b6c1-d389d7afaa73">

Reply:
Ah! I see it is because for admin I was retrieved all of the questions it was created is_enabled True whatever it is released or not .That why it is showing for all the questions.